dm/master/server.go

```diff
@@ -1944,3 +1967,36 @@ func (s *Server) workerArgsExtractor(args ...interface{}) (workerrpc.Client, str
 
 	return cli, worker, nil
 }
+
+
+// HandleHttpApis handles http apis and translate to grpc request
+func (s *Server) HandleHttpApis(ctx context.Context, mux *http.ServeMux) error {
+	// MasterAddr's format may be "host:port" or "":port"
+	items := strings.Split(s.cfg.MasterAddr, ":")
+	port := items[len(items)-1]
```

Comment:
😕 Use [`net.SplitHostPort`](https://godoc.org/net#SplitHostPort) for this?

Reply:
updated